### PR TITLE
Added Juno for CentOS 7

### PR DESCRIPTION
**Pull request by ionutbalutoiu**

### CODE REVIEW

File: file_root/cinder/volume.sls [hunk 1]
```diff
@@ -3,26 +3,52 @@
...
+    - name: "{{ salt['pillar.get']('packages:targetcli') }}"
 
-{% set count = ((salt['pillar.get']('cinder:volumes_group_size', default='30')|int)*(2**30)/4096)|int %}
+cinder_olso_db_python_install:
+  pkg:
+    - installed
+    - name: "{{ salt['pillar.get']('packages:olso_db_python') }}"
+{% endif %}
+
+{% if salt['cmd.run']('losetup -a | grep "/var/lib/cinder/cinder-volumes"') %}
```
Comment (ociuhandu) [inline file_root/cinder/volume.sls]: why drop 'cinder:volumes_path'?

File: file_root/cinder/volume.sls [hunk 2]
```diff
@@ -3,26 +3,52 @@
...
+    - run
+    - name: 'losetup -d {{ salt['pillar.get']('cinder:loopback_device') }}'
+{% endif %}
+
+{% set size = ((salt['pillar.get']('cinder:volumes_group_size')|int)*(2**30)/4096)|int %}
 dd_cmd:
   cmd:
     - run
-    - name: dd if=/dev/zero of={{ salt['pillar.get']('cinder:volumes_path', default='/var/lib/cinder/cinder-volumes') }} bs=4K count={{ (count+(0.03*count))|int }}
+    - name: dd if=/dev/zero of=/var/lib/cinder/cinder-volumes bs=4K count={{ (size+(0.03*size))|int }}
```
Comment (ociuhandu) [inline file_root/cinder/volume.sls]: ditto

File: file_root/cinder/volume.sls [hunk 3]
```diff
@@ -3,26 +3,52 @@
...
 
 cinder_volumes:
   file:
     - managed
-    - name: "{{ salt['pillar.get']('cinder:volumes_path', default='/var/lib/cinder/cinder-volumes') }}"
+    - name: "/var/lib/cinder/cinder-volumes"
```
Comment (ociuhandu) [inline file_root/cinder/volume.sls]: ditto

File: file_root/cinder/volume.sls [hunk 4]
```diff
@@ -32,28 +58,34 @@ cinder_volumes:
 losetup_cmd:
   cmd:
     - run
-    - name: losetup {{ salt['pillar.get']('cinder:loopback_device', default='/dev/loop0') }} {{ salt['pillar.get']('cinder:volumes_path', default='/var/lib/cinder/cinder-volumes') }}
+    - name: losetup {{ salt['pillar.get']('cinder:loopback_device') }} /var/lib/cinder/cinder-volumes
```
Comment (ociuhandu) [inline file_root/cinder/volume.sls]: ditto

File: pillar_root/default_template/cluster_resources.sls [hunk 5]
```diff
@@ -73,6 +73,5 @@ files:
 
 cinder:
   volumes_group_name: "cinder-volumes"
-  volumes_path: "/var/lib/cinder/cinder-volumes"
```
Comment (ociuhandu) [inline pillar_root/default_template/cluster_resources.sls:76]: volumes_path remains as parameter (see comments above)

File: file_root/top.sls [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 openstack:
   "*":
-    - generics.*
+    - generics.prerequisites
```
Comment (chvalean) [inline file_root/top.sls]: E un motiv pt specificat direct generics.prereq? Altfel ar putea intra in for %formula%, sau de ce s-a scos generics.*?

File: file_root/horizon/init.sls [hunk 7]
```diff
@@ -39,20 +49,68 @@ openstack_dashboard_ubuntu_theme_purge:
...
+
+allow_dashboard_connections:
+  cmd:
+    - run
+    - name: sed -i "s/ALLOWED_HOSTS = \[.*\]/ALLOWED_HOSTS = ['*']/" /etc/openstack-dashboard/local_settings
+    - unless: egrep "ALLOWED_HOSTS = \['\*'\]" /etc/openstack-dashboard/local_settings
+    - require:
+      - pkg: openstack_dashboard_install
+
+enable_firewall_http_access:
```
Comment (chvalean) [inline file_root/horizon/init.sls:87]: Se poate elimina cand iptables rules vor fi applicate separate.

File: file_root/neutron/init.sls [hunk 8]
```diff
@@ -16,13 +28,13 @@ neutron_conf_file:
       - ini: neutron_conf_file
   ini: 
     - options_present
-    - name: "{{ salt['pillar.get']('conf_files:neutron', default='/etc/neutron/neutron.conf') }}"
+    - name: "{{ salt['pillar.get']('conf_files:neutron') }}"
     - sections: 
         DEFAULT: 
           auth_strategy: keystone
           rpc_backend: neutron.openstack.common.rpc.impl_kombu
```
Comment (chvalean) [inline file_root/neutron/init.sls]: Pt Juno rpc_backend e rabbit, trebuie folosit variante de if pt juno/icehouse

File: file_root/neutron/openvswitch.sls [hunk 9]
```diff
@@ -33,71 +37,120 @@ l2_agent_neutron_config_file:
...
-          rabbit_password: {{ salt['pillar.get']('rabbitmq:guest_password', default='Passw0rd') }}
+{% endif %}
+          rabbit_host: "{{ get_candidate('queue.%s' % salt['pillar.get']('queue_engine')) }}"
+          rabbit_password: {{ salt['pillar.get']('rabbitmq:guest_password') }}
           core_plugin: ml2
           service_plugins: router
           allow_overlapping_ips: True
-          verbose: True
         keystone_authtoken: 
           auth_uri: "http://{{ get_candidate('keystone') }}:5000"
```
Comment (chvalean) [inline file_root/neutron/openvswitch.sls]: auth_uri pt Juno e /v2.0, trebuie integrat in if/else in functie de release

File: file_root/neutron/init.sls (hunk 8 above)
Comment (ociuhandu) [inline file_root/neutron/init.sls]: rpc_backend should not be related in any way to the openstack version. You can use any of them, provided that it is configured ok. Please make sure there is no relation between the openstack version and the rpc backend.

Sent from mobile

On 31 ian. 2015, at 18:56, Chris Valean <notifications@github.com<mailto:notifications@github.com>> wrote:

In file_root/neutron/init.slshttps://github.com/cloudbase/salt-openstack/pull/3#discussion_r23890253:

> @@ -16,13 +28,13 @@ neutron_conf_file:
>        - ini: neutron_conf_file
>    ini:
>      - options_present
> -    - name: "{{ salt['pillar.get']('conf_files:neutron', default='/etc/neutron/neutron.conf') }}"
> -    - name: "{{ salt['pillar.get']('conf_files:neutron') }}"
>   - sections:
>      DEFAULT:
>        auth_strategy: keystone
>        rpc_backend: neutron.openstack.common.rpc.impl_kombu

Pt Juno rpc_backend e rabbit, trebuie folosit variante de if pt juno/icehouse

**Pull request merged**